### ISSUE TITLE
When Servo isn't the window directly in front, scroll is stuck in previously scrolled layer

### ISSUE THREAD

**Issue by gterzian** [servo/servo#14290]
On Mac, I've noticed that, when Servo isn't the window 'in front', then even though you can still scroll  Servo because it's still viewable, it is always the same 'layer' (in the sense of the Webrender layer) that scrolls, even if you move the cursor onto a different part of the page. 

So for example taking https://output.jsbin.com/lukiye, if you start by scrolling the main page, then move Servo to the back (for example take the terminal in front of it), while still showing Servo. You can still scroll the  main page, however when you move your cursor onto the div in the middle of it,  trying to scroll it will just scroll the page, not the child div. Same thing if you start by first scrolling the child div. 

In other words, the scrolling is 'stuck' in the layer that you happened to be scrolling the last time that Servo was still 'in front'. 

I've checked, and when Servo is no longer 'in front', the [scroll in webrender](https://github.com/servo/webrender/blob/master/webrender/src/frame.rs#L297) always receives the same 'cursor', even when you actually move your cursor around...

I've also noticed that when Servo isn't in front on start-up, you cannot scroll at all.

**Comment by jdm** [comment]
This is presumably because when the window isn't active, it doesn't receive mouse movement events, so any scroll events received would use the old mouse position to choose which layer. I wonder why it receives scroll events, though?

**Comment by paulrouget** [comment]
This is an issue we encountered before with unfocused windows.

We should do what we did for `MouseInput` https://github.com/servo/glutin/pull/97 but for `MouseWheel`.

**Comment by gterzian** [comment]
@paulrouget thanks! I'll look into it in the coming weeks...

**Comment by gterzian** [comment]
@paulrouget there are a couple of PR's for this, see above. Since you reviewed the `MouseInput` changes at the time, perhaps you want to review this also?
